src/Symfony/Component/JsonPath/Tokenizer/JsonPathTokenizer.php

```diff
@@ -28,7 +27,7 @@ final class JsonPathTokenizer
     /**
      * @return JsonPathToken[]
      */
-    public static function tokenize(JsonPath $query): array
+    public static function tokenize(JsonPath $query, ?callable $functionValidator = null): array
```

Comment:
I suggest documenting the signature of that callable. It will make it easier to contributors reading that code in the future (same in other methods where it is forwarded).